### DISCUSSION TITLE
yarn build fails with exit code 137

### DISCUSSION THREAD

**Discussion by ciberbob1987** [Part-DB/Part-DB-server#513] [Q&A] [answered]
I tried to install the latest part-db development version, but yarn build exit with error code 137. I'm running a VPS with the following characteristics:

1 virtual CPU, 1 GB RAM
Ubuntu 22.04.3 LTS
Apache2 - PHP 8.2.15

I followed all the steps from [Part-DB installation guide](https://docs.part-db.de/installation/installation_guide-debian.html). Here the output from yarn build

```
user@webserver:/var/www/partdb_new$ sudo yarn build
yarn run v1.22.19
$ encore production --progress
Running webpack ...

 WARNING  Be careful when using Encore.configureLoaderRule(), this is a low-level method that can potentially break Encore and Webpack when not used carefully.
63% building 27/28 entries 4090/4100 dependencies 797/1012 modules[BABEL] Note: The code generator has deoptimised the styling of /var/www/partdb_new/var/translations/index.js as it exceeds the max of 500KB.
92% sealing asset processing TerserPluginKilled
error Command failed with exit code 137.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Any help is greatly appreciated. Thanks.

**Comment by jbtronics** [comment, marked as answer]
I think that is an out of memory error. The JavaScript optimization steps temporarily require some larger amount of memory and I would guess that's more than your server fan provide. adding some swap memory should most likely help in the build process.

If that is not possible, then you can use the pre built artifacts GitHub actions build for every version:

https://github.com/Part-DB/Part-DB-server/actions/runs/7806583671

Basically you just need to copy the files from the public folder from the ZIP to your server.

**Reply by ciberbob1987** [reply]
Thanks for the reply. Do I always need to run yarn build or is this not necessary if I copy the files from the zip?

**Reply by jbtronics** [reply]
The github actions, run `yarn build` and save the resulting files to the ZIP file you can download. So if you just copy the files then assets are already build and you dont need to run the build command again.

However you will have to copy the new files everytime you update Part-DB

**Reply by ciberbob1987** [reply]
Thanks, both of your suggestions worked well. Here are some considerations that may help someone in the same situation.

**Swap expansion**
Swap expanded from 512 to 2048 MB. Increased memory allocated to Node.js to 2048 MB with command `export NODE_OPTIONS="--max-old-space-size=8192"` as mentioned [here](https://stackoverflow.com/questions/53230823/fatal-error-ineffective-mark-compacts-near-heap-limit-allocation-failed-javas).
For some reason, yarn build executed with sudo (`nonrootuser@server:/var/www/partdb$ sudo yarn build`) fails with error code 137. If the same command is executed by the root user (`root@server:/var/www/partdb# yarn build`) it works without any problem.

**GitHub assets**
Downloaded the last release with built assets from GitHub. Insalled following the same steps from the installation guide without running yarn build. All the other steps remain unchanged. Also this solution works without any problem.
